Docs on ALTER TABLE ALTER INDEX (#5740)
ydb-platform/ydb: ydb/docs/en/core/yql/reference/yql-core/syntax/_includes/alter_table.md

You can specify any index parameters from the [`CREATE TABLE`](../create_table#secondary_index) command.

> You can also add a secondary index using the {{ ydb-short-name }} CLI [table index](https://ydb.tech/en/docs/reference/ydb-cli/commands/secondary_index#add) command.

Reviewer: All internal links need to be:

1. relative
2. lead to a markdown file (end with `.md`)

Author: ok, I will change that. I haven't noticed that, because it is just a copy-paste of the existing sentence

You can specify any index parameters from the [`CREATE TABLE`](../create_table#secondary_index) command.

> You can also add a secondary index using the {{ ydb-short-name }} CLI [table index](../../../../reference/ydb-cli/commands/secondary_index#add) command.

Reviewer: ```suggestion
{% if backend_name == "YDB" %}

You can also add a secondary index using the {{ ydb-short-name }} CLI [table index](../../../../reference/ydb-cli/commands/secondary_index.md#add) command.

{% endif %}
```

### Altering an index {#alter-index}

Indexes have type-specific parameters that can be tuned. Global indexes, whether [synchronous]({{ concept_secondary_index }}#sync) or [asynchronous]({{ concept_secondary_index }}#async), are implemented as hidden tables, and their automatic partitioning settings can be adjusted just like [those of regular tables](#additional-alter).

{% note info %}

Currently, specifying secondary index partitioning settings during index creation is not supported in either the [`ALTER TABLE ADD INDEX`](#add-index) or the [`CREATE TABLE INDEX`](../create_table#secondary_index) clauses.

{% endnote %}

```sql

> ALTER TABLE table_name ALTER INDEX index_name SET partitioning_setting_name value;
> ALTER TABLE table_name ALTER INDEX index_name SET (partitioning_setting_name_1 = value_1, ...);

Reviewer: ```suggestion
ALTER TABLE <table_name> ALTER INDEX <index_name> SET <partitioning_setting_name> <value>;
ALTER TABLE <table_name> ALTER INDEX <index_name> SET (<partitioning_setting_name_1> = <value_1>, ...);
```

```

> * `table_name`: The name of the table whose index is to be altered.
> 
> * `index_name`: The name of the index to be altered.
> 
> * `partitioning_setting_name`: The name of the setting to be altered, which should be one of the following:
>     * [AUTO_PARTITIONING_BY_SIZE]({{ concept_table }}#auto_partitioning_by_size)
>     * [AUTO_PARTITIONING_BY_LOAD]({{ concept_table }}#auto_partitioning_by_load)
>     * [AUTO_PARTITIONING_PARTITION_SIZE_MB]({{ concept_table }}#auto_partitioning_partition_size_mb)
>     * [AUTO_PARTITIONING_MIN_PARTITIONS_COUNT]({{ concept_table }}#auto_partitioning_min_partitions_count)
>     * [AUTO_PARTITIONING_MAX_PARTITIONS_COUNT]({{ concept_table }}#auto_partitioning_max_partitions_count)

Reviewer: ```suggestion
* `<table_name>`: The table name whose index is to be modified.

* `<index_name>`: The index name to be modified.

* `<partitioning_setting_name>`: The name of the setting to be altered, which should be one of the following:
    * [AUTO_PARTITIONING_BY_SIZE]({{ concept_table }}#auto_partitioning_by_size)
    * [AUTO_PARTITIONING_BY_LOAD]({{ concept_table }}#auto_partitioning_by_load)
    * [AUTO_PARTITIONING_PARTITION_SIZE_MB]({{ concept_table }}#auto_partitioning_partition_size_mb)
    * [AUTO_PARTITIONING_MIN_PARTITIONS_COUNT]({{ concept_table }}#auto_partitioning_min_partitions_count)
    * [AUTO_PARTITIONING_MAX_PARTITIONS_COUNT]({{ concept_table }}#auto_partitioning_max_partitions_count)
```

Author: I have applied the following parts of the commit suggestion:
- <...>
- "altered" -> "modified"

However, I don't like this suggestion:
```diff
-The name of the table whose...
+The table name whose...
```
To quote ChatGPT:

Potential Misinterpretation:
- The second sentence could be (mistakenly) interpreted to mean that the name itself has an index that is going to be modified. This can happen because "table name" could be read as a standalone phrase, and "whose index is to be modified" might be misunderstood to refer to an index associated directly with the "name" rather than the "table."

This suggestion has (in my opinion) the same potential for misinterpretation:
```diff
-The name of the index to be modified.
+The index name to be modified.
```

so I haven't applied it too.

{% note info %}

These settings cannot be [reset](#additional-reset).

{% endnote %}

> * `value`: The new value for the setting. Possible values include:
>     * `ENABLED` or `DISABLED` for `AUTO_PARTITIONING_BY_SIZE` and `AUTO_PARTITIONING_BY_LOAD` settings
>     * integer of `Uint64` type for the other settings
> 
> The code in the following example enables automatic partitioning by load for the index named `title_index` and sets the minimal partition count to 5:

Reviewer: ```suggestion
* `<value>`: The new value for the setting. Possible values include:
    * `ENABLED` or `DISABLED` for the `AUTO_PARTITIONING_BY_SIZE` and `AUTO_PARTITIONING_BY_LOAD` settings
    * An integer of `Uint64` type for the other settings

#### Example

The query in the following example enables automatic partitioning by load for the index named `title_index` of table `series` and sets its minimum partition count to 5:
```

```sql
ALTER TABLE `series` ALTER INDEX `title_index` SET (
    AUTO_PARTITIONING_BY_LOAD = ENABLED,
    AUTO_PARTITIONING_MIN_PARTITIONS_COUNT = 5
);
```

### Deleting an index {#drop-index}

```DROP INDEX```: Deletes the index with the specified name. The code below deletes the index named ```title_index```.

```sql
ALTER TABLE `series` DROP INDEX `title_index`;
```

> You can also remove a secondary index using the {{ ydb-short-name }} CLI [table index](../../../../reference/ydb-cli/commands/secondary_index#drop) command.

Reviewer: ```suggestion
{% if backend_name == "YDB" %}

You can also remove a secondary index using the {{ ydb-short-name }} CLI [table index](../../../../reference/ydb-cli/commands/secondary_index.md#drop) command.

{% endif %}
```
